src/app/portlets/content-types/listing/content-types.component.ts

```diff
@@ -0,0 +1,112 @@
+import { ActionHeaderOptions } from '../../../view/components/listing-data-table/action-header/action-header';
+import { BaseComponent } from '../../../view/components/_common/_base/base-component';
+import { Component } from '@angular/core';
+import { DataTableColumn } from '../../../view/components/listing-data-table/listing-data-table.component';
+import { LoggerService } from '../../../api/services/logger.service';
+import { MessageService } from '../../../api/services/messages-service';
+import { ActivatedRoute, Router } from '@angular/router';
+
+/**
+ * List of Content Types
+ * use: listing-data-table.component
+ * @export
+ * @class ContentTypesPortletComponent
+ * @extends {BaseComponent}
+ */
+@Component({
+    selector: 'content-types',
+    templateUrl: 'content-types.component.html'
+})
+export class ContentTypesPortletComponent extends BaseComponent {
+    public item: any;
+    contentTypeColumns: DataTableColumn[];
+    private actionHeaderOptions: ActionHeaderOptions;
+
+    constructor(messageService: MessageService, private router: Router, private route: ActivatedRoute) {
+        super(['Structure-Name', 'Variable', 'Description', 'Entries', 'delete', 'Actions', 'message.structure.delete.structure.and.content',
+                'message.structure.cantdelete'], messageService);
+    }
+
+    /**
+     * Callback call from BaseComponent when the messages are received.
+     * @memberOf ContentTypesPortletComponent
+     */
+    onMessage(): void {
+        this.actionHeaderOptions = {
+            primary: {
+                command: ($event) => {
+                    this.createContentType($event);
+                },
+                model: [
+                    {
+                        command: ($event) => {
+                            this.createContentType('content', $event);
+                        },
+                        icon: 'fa-newspaper-o',
+                        label: 'Content'
```

Comment:
Really, this label are not message for the user, maybe the better is change label by name for this property.